core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -504,6 +517,11 @@ static List<ViewHistoryEntry> updateHistory(List<ViewHistoryEntry> history, Set<
         }
       }
 
+      // keep the last numVersionsToKeep in the history
```

Comment:
previously the code implied that `history` and `versions` would have the same size, but that isn't true, because the history is now only updated when `setCurrentVersionId(..)` is being called -> keep only the last `numVersionsToKeep` in the history

Reply:
@nastra, I don't think this change is a good idea. The `numVersionsToKeep` option applies to versions, not history. History is retained as long as it references valid versions.

There's no expectation that history and versions have the same size. History will have entries that are independent from versions. For example, if I run `CREATE OR REPLACE` and add a new version then I create both a version and a history entry. But if I roll back that change by setting the current version, no versions are added but the history is updated to show that the current version went from 2 to 1 (for example).

I don't think we need to update this.

Reply:
For some reason I thought `version.history.num-entries` applies to both `versions` and `history` due to having `history` in its name. In hindsight this was a wrong assumption, so I've rolled back those changes

Reply:
I'm a bit confused, I get that history and versions are independent from each other but here we want `version.history.num-entries` to apply to history no? Or is it just bad naming of the property and it should apply to versions?